### ISSUE TITLE
util-linux failed to build on CentOS release 6.8 (Final)

### ISSUE THREAD

**Issue by FrederikLauber** [Linuxbrew/homebrew-core#1480]
### Bug reports:

https://gist.github.com/77b1a0f23207ffaed1f6e1ba409f5ea5

From what I understand, there is a problem with finding the pam libraries.

Thanks in advance for any help with this problem.

**Comment by maxim-belkin** [comment]
Blocked by #1186

**Comment by maxim-belkin** [comment]
I looked more closely at the gist-log and now I'm not so sure that the problems you see are due to pam. Please fix the problems identified by running `brew doctor` and try installing again.

**Comment by FrederikLauber** [comment]
Hi,

I am not sure that I can fix all the warnings, at least one of the two is due to the way how multiple python versions are handled in CentOS. The second one, I am still tracking to see what adds the LD_Library variable.

Maybe I missread the gist-log but does it not complain about -lpam and -lpam_misc, being unable to find the two?

**Comment by maxim-belkin** [comment]
> I am not sure that I can fix all the warnings

you have to. one of the problems is that python is not found. this is probably due to those system oython artifacts. try using alternative location for Linuxbrew.

**Comment by maxim-belkin** [comment]
what is your PATH?

**Comment by FrederikLauber** [comment]
``echo $PATH
/home/p91/.linuxbrew/sbin:/package/host/localhost/python-2.7/bin:/home/p91/.linuxbrew/bin:/package/host/localhost/python-2.7/bin:/package/host/localhost/gcc-5/bin:/home/p91/.gem/ruby/2.3.0/bin:/package/host/localhost/ruby-2.3.1/bin:/package/host/localhost/php-7.0.5/bin:/package/host/localhost/perlbrew/bin:/package/host/localhost/gnutls-3.3/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin:/var/qmail/bin:/home/p91/bin:/home/p91/bin
``

**Comment by FrederikLauber** [comment]
I got the first warning removed by unseting the variable.
Apparently it is set in the setup of my hoster due to their support of toast.

The second warning I am afraid cannot be removed. There are multiple python versions installed and hence there are also multiple-config files.

I also worked on clearing the Path up but it does not change anything in regards to the error I am seeing.

**Comment by maxim-belkin** [comment]
To start, remove `/usr/local/bin` and `/usr/local/sbin` from your PATH.
When I asked for PATH I was hoping to get the result of `echo -e ${PATH//:/\\n}`.

**Comment by FrederikLauber** [comment]
Hi,

thanks for sticking with me.

Got it to:
``[p91@cygnus ~]$ brew doctor
Your system is ready to brew.``

PATH:
``[p91@cygnus ~]$ echo -e ${PATH//:/\\n}
/home/p91/.linuxbrew/bin
/home/p91/.linuxbrew/sbin
/package/host/localhost/perlbrew/bin
/bin
/usr/bin
/usr/sbin
/sbin
/home/p91/bin
/home/p91/bin
``

But the problem remains

**Comment by maxim-belkin** [comment]
@FrederikLauber use three backticks (```) when copy-pasting code

**Comment by maxim-belkin** [comment]
can you post gist-logs now?

**Comment by FrederikLauber** [comment]
https://gist.github.com/0b0b69b3cca139294fd766e0f7dd74d6

**Comment by maxim-belkin** [comment]
Yes, I'm trying to finish up `linux-pam`

**Comment by sjackman** [comment]
One possible workaround is to install Linuxbrew in `/home/linuxbrew/.linuxbrew` so that you can use the precompiled binary bottle for `util-linux`.

Does adding `configure --disable-pam` (if such an option exists) to `util-linux` help this issue?

**Comment by FrederikLauber** [comment]
```configure --disable-pam``` does not help.

As I am only a user on the machine in question, I cannot switch over to ```/home/linuxbrew/.linuxbrew```

**Comment by sjackman** [comment]
You can if you like ask your system administrator to create a `linuxbrew` user account.
You can also try `brew install --force-bottle util-linux`. It may work, but no promises.

**Comment by maxim-belkin** [comment]
> ```configure --disable-pam``` does not help.

can you share gist-logs for that build?

**Comment by sjackman** [comment]
Note that I haven't check that there is `--disable-pam` or `--without-pam` option. You'll have to check `./configure --help`.

**Comment by carlorosati** [comment]
I already found the workaround in https://github.com/Linuxbrew/homebrew-core/issues/1018#issuecomment-261057772

**Comment by sjackman** [comment]
For the record, the workaround to compile `util-linux` without PAM is:
```sh
configure --disable-chfn-chsh --disable-login --disable-nologin --disable-su --disable-runuser
```

**Comment by sjackman** [comment]
Duplicate of #1018.

**Comment by sjackman** [comment]
Another work around is `brew install --force-bottle util-linux`
